cmake/build_configurations/mysql_release.cmake

```diff
@@ -187,30 +187,32 @@ IF(UNIX)
   # Default GCC flags
   IF(CMAKE_COMPILER_IS_GNUCC)
     SET(COMMON_C_FLAGS               "-g -fno-omit-frame-pointer -fno-strict-aliasing  -Wno-uninitialized")
-    SET(CMAKE_C_FLAGS_DEBUG          "-O ${COMMON_C_FLAGS}")
-    SET(CMAKE_C_FLAGS_RELWITHDEBINFO "-O3 ${COMMON_C_FLAGS}")
+    SET(CMAKE_C_FLAGS_DEBUG          "${CMAKE_C_FLAGS_DEBUG} ${COMMON_C_FLAGS}")
+    SET(CMAKE_C_FLAGS_RELEASE        "${CMAKE_C_FLAGS_RELEASE} -O3 ${COMMON_C_FLAGS}")
+    SET(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -O3 ${COMMON_C_FLAGS}")
```

Comment:
Better string(REGEX REPLACE) to replace -O2 with -O3 .Or, there will be both -O2 and -O3 in the compilation. We know, that last option happens to win in GCC, but only one -O is cleaner.